frontend/src/app/questions/questions.component.html

```diff
@@ -0,0 +1,158 @@
+            datakey="id"
+            [tableStyle]="{ 'table-layout': 'fixed' }"
+            [paginator]="true"
+            [rows]="5"
+            [rowsPerPageOptions]="[5, 10, 20]"
+            styleClass="p-datatable-gridlines-striped">
+            <ng-template pTemplate="caption">
+                <div class="flex">
+                    <h3 class="m-0">Manage Questions</h3>
+                </div>
+            </ng-template>
+            <ng-template pTemplate="header" let-columns>
+                <tr>
+                    <th style="width: 10%"><p-tableHeaderCheckbox /></th>
+                    <th pSortableColumn="id" style="width: 10%">Id <p-sortIcon field="id" /></th>
+                    <th style="width: 10%">Title</th>
+                    <th style="width: 40%">Description</th>
+                    <th style="width: 10%">Topics</th>
+                    <th style="width: 10%">Difficulty</th>
+                    <th style="width: 10%"></th>
+                </tr>
+            </ng-template>
+            <ng-template pTemplate="body" let-question>
+                <tr>
+                    <td>
+                        <p-tableCheckbox [value]="question" />
+                    </td>
+                    <td>{{ question.id }}</td>
+                    <td>{{ question.title }}</td>
+                    <td>{{ question.description }}</td>
+                    <td>{{ question.topics.join(', ') }}</td>
+                    <td>{{ question.difficulty }}</td>
+                    <td>
+                        <p-button
+                            label="Edit"
+                            severity="primary"
+                            icon="pi pi-file-edit"
+                            class="mr-2"
+                            [text]="true"
+                            (onClick)="editQuestion(question)" />
+                    </td>
+                </tr>
+            </ng-template>
+        </p-table>
+    </ng-container>
+    <p-dialog header="Header" [(visible)]="isDialogVisible" [modal]="true" [style]="{ width: '25rem' }">
+        <ng-template pTemplate="header">
+            <div class="inline-flex align-items-center justify-content-center gap-2">
+                <span class="font-bold white-space-nowrap"> Create new question </span>
+            </div>
+        </ng-template>
+
+        <form [formGroup]="questionFormGroup">
+            <div class="field grid col-12">
+                <label for="title">Title</label>
+                <input
+                    formControlName="textTitle"
+                    type="text"
+                    pInputText
+                    id="title"
+                    required
+                    class="text-base text-color surface-overlay p-2 border-1 border-solid surface-border border-round appearance-none outline-none focus:border-primary w-full" />
+                <small class="p-error" *ngIf="submitted && (question.title === undefined || !question.title)">
```

Comment:
Shall we use the new Angular `@if() {...}` syntax instead?

Also we can use the form validators instead of manual checks here. (see [register.component.html](https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g03/blob/7662753d36e536485635c702135b3c755ce1bc85/frontend/src/app/account/register.component.html#L41) and [register.component.ts](https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g03/blob/7662753d36e536485635c702135b3c755ce1bc85/frontend/src/app/account/register.component.ts#L62))